### PR TITLE
add subscriptions to metadata endpoint HTTP api

### PR DESCRIPTION
**Pull request by famarting**
Signed-off-by: Fabian Martinez <46371672+famarting@users.noreply.github.com>

# Description
fixes https://github.com/dapr/dapr/issues/5510

Implements the changes required for exposing all application subscription through the metadata endpoint, reusing the runtime logic.

## Issue reference

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: dapr/proto/runtime/v1/dapr.proto [hunk 1]
```diff
@@ -541,6 +542,24 @@ message RegisteredComponents {
   repeated string capabilities = 4;
 }
 
+message PubsubSubscription {
+  string pubsub_name = 1;
+  string topic = 2;
+  map<string,string> metadata = 3;
+  PubsubSubscriptionRules rules = 4;
+  string dead_letter_topic = 5;
+  bool programmatic_subscription = 6;
```
Comment (yaron2) [inline dapr/proto/runtime/v1/dapr.proto:551]: Why is this needed?

Comment (famarting) [reply dapr/proto/runtime/v1/dapr.proto:551]: the flag prgrammatic_subscription? I thought it would be useful to differentiate how the subscription was generated

**Comment by yaron2** [comment]
> If the intent is to fetch component configuration data

This is not the intent here.

File: dapr/proto/runtime/v1/dapr.proto (hunk 1 above)
Comment (yaron2) [reply dapr/proto/runtime/v1/dapr.proto:551]: I would rather remove this to drive consistency across subscription methods.

**Review by yaron2 — CHANGES_REQUESTED** [review]

**Review by yaron2 — APPROVED** [review]
LGTM
